src/radical/entk/execman/rp/task_processor.py

```diff
@@ -177,32 +177,19 @@ def resolve_arguments(args, placeholders, logger):
 #
 def resolve_tags(task, parent_pipeline_name, placeholders):
 
-    # entk only handles co_location tags.  If tags are given as strings, they
-    # get translated into `{'colocate': '<tag>'}`.  Tags passed as dictionaries
-    # are checked to conform with above form.
-    #
-    # In both cases, the tag string is expanded with the given placeholders.
+    tags = dict(task.tags or {'colocate': task.uid})
 
-    tags = task.tags if task.tags else {'colocate': task.uid}
+    pnames = list(placeholders)
+    pnames.remove(parent_pipeline_name)
 
     colo_tag = tags['colocate']
-
-    # Check self pipeline first
-    for sname in placeholders[parent_pipeline_name]:
-        if colo_tag in placeholders[parent_pipeline_name][sname]:
-            return {'colocate': placeholders[parent_pipeline_name][sname][colo_tag]['uid']}
-
-    for pname in placeholders:
-
-        # skip self pipeline this time
-        if pname == parent_pipeline_name:
-            continue
-
+    for pname in [parent_pipeline_name] + pnames:
```

Comment:
this ensures that the `parent_pipeline_name` is always included in the loop.  Are we sure that this is always in the `placeholders` dict?  Or should we add a check?  I tried to dig through the placeholder creation, but oh boy, that code is a challenge.  What's this `__by__name__ thingie?  Anyway, that's unrelated to this PR...

Reply:
with this one I didn't change the logic, just tried to simplify it a little - to have one pass through all values and ensure that `parent_pipeline_name` comes the first. Same as above - also didn't try to dig further here